### PR TITLE
Brain performance and timings

### PR DESCRIPTION
**Pull request by nk3nny**
### Brain performance and tuned timings: 
Adds exit condition on 'lambs_danger_forceMove' in brainForced
Improved performance of brainEngage by having zero-time exit variables
Improved performance by skipping unnecessary vector additions to suppression

### CODE REVIEW

File: addons/main/script_macros.hpp [hunk 1]
```diff
@@ -18,7 +18,7 @@
     _unit \
 }
 
-// #define DISABLE_COMPILE_CACHE
+#define DISABLE_COMPILE_CACHE
```
Comment (jokoho48) [inline addons/main/script_macros.hpp:21]: you forgot this here D:

Comment (nk3nny) [reply addons/main/script_macros.hpp:21]: Haha, yeah noticed. blargh

File: addons/danger/scripts/lambs_danger.fsm [hunk 2]
```diff
@@ -230,7 +245,7 @@ class FSM
                         itemno = 9;
                         init = /*%FSM<STATEINIT""">*/"" \n
                          "// keep attacking" \n
-                         "if (_dangerCause in [8, 9] && {_dangerCausedBy call lambs_main_fnc_isAlive}) then {" \n
+                         "if (_dangerCause in [8, 9] && {alive _dangerCausedBy}) then {" \n
```
Comment (diwako) [inline addons/danger/scripts/lambs_danger.fsm:248]: Why the change to just a simple alive check?

Comment (nk3nny) [reply addons/danger/scripts/lambs_danger.fsm:248]: Two reasons: 
1. Performance (slight)
2. ACE3

ACE3 sets incapacitated units as functionally _invisible_ to the AI. However, that invisibility does not mean the AI isn't being told by its native FSM to try to find and shoot them.  Hence, a simple checks allows our dangerFSM to run _on_ that target-- instead of having them uselessly flail about. 

As it pertains to vehicles, this means a range of interesting and immersive actions.  Such as generally suppressing the location of suspected enemies and _jinking_ out of danger of the same. :)

File: addons/danger/functions/fnc_isForced.sqf [hunk 3]
```diff
@@ -10,17 +10,12 @@
...
  *
  * Example:
- * [bob] call lambs_danger_fnc_isForced;
+ * bob call lambs_danger_fnc_isForced;
  *
  * Public: No
 */
-params ["_unit"];
-fleeing _unit
-|| {_unit getVariable [QGVAR(disableAI), false]}
```
Comment (jokoho48) [inline addons/danger/functions/fnc_isForced.sqf:19]: do we deprecating this variable? or is it handled elsewhere?

**Review by jokoho48 — APPROVED** [review]

File: addons/danger/scripts/lambs_danger.fsm (hunk 2 above)
Comment (nk3nny) [reply addons/danger/scripts/lambs_danger.fsm]: Reverted
